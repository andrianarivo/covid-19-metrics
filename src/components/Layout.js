import { Outlet } from 'react-router-dom';
import NavBar from './NavBar';

export default function Layout() {
  return (
    <>
      <NavBar />
      <Outlet />
    </>
  );
}
