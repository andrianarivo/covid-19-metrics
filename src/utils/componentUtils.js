import asia from '../images/asia.png';
import africa from '../images/africa.png';
import northAmerica from '../images/north_america.png';
import southAmerica from '../images/south_america.png';
import oceania from '../images/oceania.png';
import europe from '../images/europe.png';

const getImage = (continent) => {
  switch (continent.toLowerCase()) {
    case 'asia':
      return asia;
    case 'africa':
      return africa;
    case 'north america':
      return northAmerica;
    case 'south america':
      return southAmerica;
    case 'australia-oceania':
      return oceania;
    case 'europe':
      return europe;
    default:
      return asia;
  }
};

export default getImage;
