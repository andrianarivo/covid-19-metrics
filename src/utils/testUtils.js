import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';

export function withRouter(children) {
  return (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );
}

export function withProvider(children, mockStore = store) {
  return (
    <Provider store={mockStore}>
      {children}
    </Provider>
  );
}
