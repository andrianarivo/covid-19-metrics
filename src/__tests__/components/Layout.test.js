import { create } from 'react-test-renderer';
import { Route, Routes } from 'react-router-dom';
import { withRouter } from '../../utils/testUtils';
import Layout from '../../components/Layout';

describe('Testing Layout component', () => {
  const renderLayout = () => withRouter(
    <Routes>
      <Route path="/" element={<Layout />}>
        <Route index element={<main>Mock Home</main>} />
      </Route>
    </Routes>,
  );

  test('renders correctly', async () => {
    const tree = create(renderLayout());
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
