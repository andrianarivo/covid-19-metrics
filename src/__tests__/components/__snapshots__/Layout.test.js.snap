// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Layout component renders correctly 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          data-testid="home-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
    </ul>
  </nav>,
  <main>
    Mock Home
  </main>,
]
`;
